### ISSUE TITLE
Spaces indentation improvement - align indenting left/right to rounded indentation level

### ISSUE THREAD

**Issue by iwek7** [godotengine/godot#28724]
Hello,

I'd like to propose change to current indentation mechanics in GDScript editor. There are some cases that are causing issues for me, consider following situations where indent is set to 4 spaces:

![image](https://user-images.githubusercontent.com/9964886/57248310-58b63780-7042-11e9-8ee1-ce042ff2e771.png)
As you can notice I accidentally hit space on line 15 which gives this line total indent of 5 spaces while the rest of code is indented to 4. Okey, I hit backspace (or shitf+tab) to delete this space - but I actually end up with just one space like this:
![image](https://user-images.githubusercontent.com/9964886/57248464-b2b6fd00-7042-11e9-820c-49f2faae4f5a.png)

Obviously i'd expect deletion to go stop at 4 spaces to align to closest full indentation level - this is the way it works in most code editors (for instance in sublime).

The same goes for indenting - if I have cursor set after one space hitting tab editor should move cursor to next full indentation level of 4 - now it just adds four spaces giving line 5 spaces of indentation.

Another thing that could be changed is clicking backspace where there are less spaces than indentation level for instance:

![image](https://user-images.githubusercontent.com/9964886/57249882-62419e80-7046-11e9-86a4-8ae1f834687a.png)

I'd expect whole indentation level removed here whereas spaces are removed one by one (this is special case due to ifs logic in code_editor.cpp code).

To sum up the changes I propose are:
1) Hitting tab would move cursor to next full indentation level instead of adding fixed number of spaces
2) Hitting backspace would remove spaces up to next full indentation level
3) Hitting shift+tab would remove spaces up to next full indentation level
4) Hitting backspace when there are less spaces than indentation size would remove spaces up to beginning of the line

Since there is lot of going on I decided to create Issue to ask about your opinion but I have PR work in progress - let me know if this feature is wanted and I'll provide code for it.

Ps.: There seems to also be separate code branch for indenting when selection is not empty, this needs to be adjusted accordingly too

**Comment by akien-mga** [comment]
Fixed by #28811.
